frontend/rust-lib/flowy-core/src/deps_resolve/user_deps.rs

```diff
@@ -61,9 +62,14 @@ impl UserWorkspaceService for UserWorkspaceServiceImpl {
   }
 
   fn did_delete_workspace(&self, workspace_id: String) -> FlowyResult<()> {
-    self
+    // The remove_indices_for_workspace should not block the deletion of the workspace
+    // Log the error and continue
+    if let Err(err) = self
       .folder_manager
-      .remove_indices_for_workspace(workspace_id)?;
+      .remove_indices_for_workspace(workspace_id)
+    {
+      info!("Error removing indices for workspace: {}", err);
+    }
```

Comment:
@Xazin FYI, the `remove_indices_for_workspace` should not block the deletion of the workspace if it has actually been deleted successfully.

However, the `remove_indices_for_workspace` might sometimes throw an error, possibly with a 1 in 5 chance.

```
{"msg":"[Flutter]: delete workspace error: code: FolderIndexManagerUnavailable\nmsg: FolderIndexManager or its dependencies are unavailable\n","time":"07-30 21:33:30","target":"dart_ffi"}
```